### ISSUE TITLE
Get GUI tests running on Travis

### ISSUE THREAD

**Issue by ixjlyons** [ucdrascal/axopy#28]
I don't care *all* that much about GUI tests themselves, but they do serve some purpose as they test that the installation process for PyQt5 works end to end (which is one of the few dependencies that's not necessarily trivial to get installed and running). Currently, running the tests in `tests/test_gui.py` on Travis results in a segfault. I've been playing around with docker containers and have currently narrowed it down to the following image to get the tests to run without errors:

```dockerfile
FROM ubuntu:16.04

RUN apt-get update && \
    apt-get install -y python3 python3-pip python3-setuptools xvfb xorg libglib2.0

ADD . /root/src/axopy

RUN cd /root/src/axopy && \
    pip3 install -r requirements.txt && \
    pip3 install -r requirements-dev.txt && \
    pip3 install -e .
```

Note that I actually installed xorg interactively in the container and I couldn't figure out how to install it without prompting for a language and keyboard layout (probably just have to set that up manually before installing).

So starting with a minimal Ubuntu install, you just need Python itself installed (could instead be done with conda) and a few packages that you are *very* likely to already have if you're using it as a desktop system (except maybe xvfb). Unfortunately they're pretty huge packages (pull in quite a few dependencies) so I'd prefer to figure out what's required more minimally.

**Comment by ixjlyons** [comment]
Useful little command to run inside a bare Ubuntu16.04 container to see what is going wrong (because pytest and/or pytest-qt output doesn't seem to capture it):

```bash
python3 -c "from PyQt5.QtWidgets import QApplication; QApplication([])"
```

That will raise something like this:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
ImportError: libgthread-2.0.so.0: cannot open shared object file: No such file or directory
```

**Comment by ixjlyons** [comment]
The traceback in the above comment (`libgthread-2.0.so.0: cannot open shared object file...`) indicates there's a system library being looked for (i.e. not provided with the PyQt5 wheel) on Linux. I just found that the file `libgthread-2.0.so` is provided by the libglib2.0-0 package. It pulls in a number of dependencies but it's not that bad. xorg is the one to start narrowing down now.

**Comment by ixjlyons** [comment]
Here's the best I've got so far. 78 packages installed on top of the base 16.04 image and all tests run successfully. It's possible only `xvfb` is needed to get things running on Travis, because I think it generally sets up a lot of xorg stuff.

```dockerfile
RUN apt-get update -y && \
    apt-get install -y --no-install-recommends \
        python3 python3-pip python3-setuptools \
        libglib2.0-0 libgl1-mesa-glx libfontconfig1 \
        libxi6 libxrender1 libdbus-1-3 xvfb

ADD . /root/src/axopy

RUN cd /root/src/axopy && \
    pip3 install --upgrade pip && \
    pip3 install -r requirements.txt && \
    pip3 install -r requirements-dev.txt && \
    pip3 install -e .
```

**Comment by ixjlyons** [comment]
Fixed by #29
